x/btcstaking/types/finality_provider.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+// TODO: Rethink how to check whether an fp is a Babylon Genesis fp or not.
+// Checking through the ChainID set by context is very brittle and will
+// certainly lead to issues down the road.
+func (fp *FinalityProvider) SecuresBabylonGenesis(ctx sdk.Context) bool {
```

Comment:
The other way (that is less brittle) is to leave `BsnID` empty for Babylon FPs in KV store. This at least avoids bugs that exploit chain ID in context.

Reply:
Yep, we can examine this in a separate PR. Leaving this as is now as it requires discussion.

Reply:
>  though this will influence the indext bsnId -> fkpKeys

Babylon FP shall not be in this index, right?